auth-service/src/main/java/com/petclinic/authservice/presentationlayer/User/UserController.java

```diff
@@ -104,6 +104,10 @@ public ResponseEntity<UserPasswordLessDTO> updateUserRole (@PathVariable String
         return ResponseEntity.ok().body(userService.updateUserRole(userId, roleChanged, token));
     }
 
+    @PutMapping("/{userId}")
+    public ResponseEntity<User> updateUser (@PathVariable String userId, @RequestBody UserPasswordLessDTO userPasswordLessDTO) {
+        return ResponseEntity.ok().body(userService.updateUser(userId, userPasswordLessDTO));
+    }
```

Comment:
```suggestion
   public Mono<UserPasswordLessDTO> updateUser(@PathVariable String userId, @RequestBody @Valid UserPasswordLessDTO model, @CookieValue("Bearer") String jwtToken) {
    return authServiceClient.updateUser(userId, model, jwtToken);
}
```
You can add validation for the UserPasswordLessDTO model in the updateUser method to ensure data integrity.